src/Renderer/Shader/TileFS.glsl

```diff
@@ -122,21 +122,22 @@ void main() {
                             textureIndex,
                             projWGS84 ? vUv_WGS84 : uvPM);
 
-                        if (layerColor.a > 0.0) {
+                        bool isVisible = layerColor.a > 0.0;
+                        // if first texture is transparent, we need to force blending with first background color (CWhite)
+                        if ((isVisible && paramsA.w > 0.0) || !validTexture) {
                             validTexture = true;
-                            float lum = 1.0;
-
-                            if(paramsA.z > 0.0) {
-                                float a = max(0.05,1.0 - length(layerColor.xyz-CWhite.xyz));
-                                if(paramsA.z > 2.0) {
-                                    a = (layerColor.r + layerColor.g + layerColor.b)*0.333333333;
-                                    layerColor*= layerColor*layerColor;
-                                }
-                                lum = 1.0-pow(abs(a),paramsA.z);
+                            if(paramsA.z > 2.0) {
+                                float a = max(0.05,1.0 - length(layerColor.xyz - CWhite.xyz));
```

Comment:
I'm not really sure of what is happening when paramsA.z is > 0.0 or > 2.0 but: are you sure that you shouldn't de-premultiply-alpha before tweaking the color?

Actually, you could do something like:
```glsl
if (paramsA.z > 2.0) {
    layerColor = applyXXXEffect(layerColor / layerColor.a);
} else if (paramsA.z > 0.0) {
    layerColor = applyYYYEffect(layerColor / layerColor.a);
}
```
to improve readability. WDYT?

Reply:
yes, we need to de-premultiply and re-premultiply